xlive/Blam/Engine/math/real_math.h

```diff
@@ -475,6 +480,22 @@ if (!valid_real_normal3d(normal))				\
 }												\
 (void)0
 
+#define assert_valid_real_point3d(point)		\
+if (!valid_real_point3d(point))					\
+{												\
+	const char* string = csprintf(g_temporary, NUMBEROF(g_temporary), "%s: assert_valid_real_point3d(%f, %f, %f)", #point, (*point).x, (*point).y, (*point).z);	\
+	DISPLAY_ASSERT(string);						\
+}												\
+(void)0
+
+#define assert_valid_real_vector3d(vector)		\
+if (!valid_real_vector3d(vector))				\
+{												\
+	const char* string = csprintf(g_temporary, NUMBEROF(g_temporary), "%s: assert_valid_real_vector3d(%f, %f, %f)", #vector, (*vector).i, (*vector).j, (*vector).k);	\
```

Comment:
`vector->i` ?

Reply:
Ex. `&forward`, you wont be able to do `&forward->i`